src/charm.py

```diff
@@ -1406,6 +1407,12 @@ def _reconciliation(self, _event: EventBase) -> None:
             self.unit.status = WaitingStatus("Waiting for storage")
             _event.defer()
             return
+        # Ensure the data directory for WordPress is writable for the WordPress user.
+        self._container().exec(
+            f"chown {self._WORDPRESS_USER}:{self._WORDPRESS_GROUP} -R {self._WP_UPLOADS_PATH}".split(
```

Comment:
I think we might just need to change the ownership of the mounting directory, i.e. without `-R`. Since recursively change the ownership of all files can be slow if there are a lot files in the directory.

Reply:
Its definitely a good point but since the _WORDPRESS_USER was changed recently from www-data to _daemon_, I was afraid we might have a case where files would be with the previous permission. What do you think?

Reply:
BTW there is no wait for output, do you think it will affect the charm anyway?

Reply:
I think the `wait` might be necessary, `exec` without `wait` might be undefined behavior.

Yes, you are correct the upgrade might cause some problem. Could we run the recursive version only on upgrade event so we can keep the reconciliation process as light as possible as the reconciliation can run multiple times during the charm's lifecycles https://github.com/canonical/wordpress-k8s-operator/issues/145

Reply:
I see. I made some changes now, what do you think?

Reply:
👍